dynamic_rest/paginator.py

```diff
@@ -0,0 +1,96 @@
+# adapter from Django's django.core.paginator
+# adds support for the "exclude_count" parameter
+
+from math import ceil
+
+import inspect
+from django.utils.functional import cached_property
+from django.core.paginator import Paginator
+from django.utils.inspect import method_has_no_args
+
+try:
+    from django.utils.translation import gettext_lazy as _
+except ImportError:
+    def _(x):
+        return x
+
+
+class InvalidPage(Exception):
+    pass
+
+
+class PageNotAnInteger(InvalidPage):
+    pass
+
+
+class EmptyPage(InvalidPage):
+    pass
+
+
+class DynamicPaginator(Paginator):
+
+    def __init__(self, object_list, per_page, orphans=0,
+                 allow_empty_first_page=True, exclude_count=False):
+        self.object_list = object_list
+        self._check_object_list_is_ordered()
+        self.per_page = int(per_page)
+        self.orphans = int(orphans)
+        self.allow_empty_first_page = allow_empty_first_page
+        self.exclude_count = exclude_count
+
+    def validate_number(self, number):
+        """Validate the given 1-based page number."""
+        try:
+            number = int(number)
+        except (TypeError, ValueError):
+            raise PageNotAnInteger(_('That page number is not an integer'))
+        if number < 1:
+            raise EmptyPage(_('That page number is less than 1'))
+        if self.exclude_count:
+            # skip validating against num_pages
+            return number
+        if number > self.num_pages:
+            if number == 1 and self.allow_empty_first_page:
+                pass
+            else:
+                raise EmptyPage(_('That page contains no results'))
+        return number
+
+    def page(self, number):
+        """Return a Page object for the given 1-based page number."""
+        number = self.validate_number(number)
+        bottom = (number - 1) * self.per_page
+        top = bottom + self.per_page
+        if self.exclude_count:
+            # always fetch one extra item
+            # to determine if more pages are available
+            top = top + 1
+        else:
+            # skip validating against count
```

Comment:
this comment line should be in the `if` part of this conditional or can be removed altogether

Reply:
done